src/main/java6/net/finmath/marketdata/model/curves/Curve.java

```diff
@@ -493,4 +486,12 @@ public CurveBuilderInterface getCloneBuilder() throws CloneNotSupportedException
 		CurveBuilder curveBuilder = new CurveBuilder(this);
 		return curveBuilder;
 	}
-}
+	
+	@Override
+	public String toString() {
+		String pointsAsString = "";
+		for (Point point : points)
+			pointsAsString += "\n[" + Schedule.getDateFromDouble(getReferenceDate(),point.time) + ": " + valueFromInterpolationEntity(point.value, point.time) + "]";

```

Comment:
Oh I did not know that this would ever intetendedly be null (though as always a null-pointer check would have also made sense solely to cover the unintented cases). Could you ecplain the use-case of having refDate=null in bit more detail? How about checking for refDate=null in the toString and then either have the pretty printing or the Standard one. I can only say that having the actual dates and not the doubles helps me a lot when debugging the code -  so I'm a bit reluctant to go without that feature